main.tf

```diff
@@ -22,7 +22,9 @@ resource "random_id" "random_role_id_suffix" {
 }
 
 locals {
-  base_role_id = "osLoginProjectGet"
+  base_role_id          = "osLoginProjectGet"
+  service_account_email = var.create_service_account ? google_service_account.bastion_host[0].email : format("%s@%s.iam.gserviceaccount.com", var.service_account_name, var.project)
```

Comment:
Instead of assuming the service account lives in the same project as the bastion, let's replace the `create_service_account` variable with `service_account_email` and assume the service account has the correct permissions (we'd also need to specify which permissions are necessary in the variable description).

Reply:
that makes sense